packages/yoga/src/Button/web/Icon.jsx

```diff
@@ -0,0 +1,125 @@
+import React, { forwardRef } from 'react';
+import { oneOfType, func, node, bool } from 'prop-types';
+import styled from 'styled-components';
+import { hexToRgb } from '@gympass/yoga-common';
+
+import Button from './Button';
+
+const IconStyled = styled(Button)`
+  ${({
+    large,
+    inverted,
+    secondary,
+    theme: {
+      yoga: {
+        components: { button },
+      },
+    },
+  }) => {
+    const state = secondary ? 'secondary' : 'primary';
+
+    return `
+      width: ${
+        large ? button.types.icon.size.large : button.types.icon.size.default
+      }px;
+      height: ${
+        large ? button.types.icon.size.large : button.types.icon.size.default
+      }px;
+      padding: 0;
+      border-radius: 50%;
+
+      svg {
+        margin: 0;
+        width: ${
+          large ? button.types.icon.svg.large : button.types.icon.svg.default
+        }px;
+        height: ${
+          large ? button.types.icon.svg.large : button.types.icon.svg.default
+        }px;
+        fill: ${button.types.contained.font.default.color};
+ 
+        transition: fill 0.2s;
+      }
+
+      &:disabled,
+      &:not([disabled]):hover,
+      &:not([disabled]):focus,
+      &:not([disabled]):active {
+        box-shadow: unset;
+      }
+
+      &:not([disabled]):hover  {
+        color: ${hexToRgb(button.types.link.font[state].color, 0.5)};
+      }
+
+      &:not([disabled]):focus, &:not([disabled]):active {
+        color: ${hexToRgb(button.types.link.font[state].color, 0.75)};
+      }
+
+      &:disabled {
+        color: ${button.types.link.font.disabled.color};
+      }
+
+      ${
+        inverted
+          ? `
+              background-color: ${button.types.contained.font.default.color};
+              color: ${button.types.contained.backgroundColor[state].default};
+      
+              svg {
+                fill: ${button.types.contained.backgroundColor[state].default};
+              }
+      
+              &:active {
+                background-color: ${hexToRgb(
+                  button.types.contained.font.default.color,
+                  0.75,
+                )};
+                color: ${button.types.contained.backgroundColor[state].pressed};
+      
+                svg {
+                  fill: ${
+                    button.types.contained.backgroundColor[state].pressed
+                  };
+                }
+              }
+      
+              &:not([disabled]):hover, &:not([disabled]):focus {
+                box-shadow: 0 4px 8px ${hexToRgb(
+                  button.types.contained.font.default.color,
+                  0.45,
+                )};
+              }
+            `
+          : ''
+      }
+    `;
+  }}
+`;
+
+const ButtonIcon = forwardRef(({ icon: Icon, ...props }, ref) => (
+  <IconStyled ref={ref} {...props}>
+    <Icon />
```

Comment:
@ericcleao Do you think we should use `Icon`, from Yoga, to handle this component?
It would change how the `Button.Icon.theme` has been set. 

Do you think it makes sense and is a valid effort?

Reply:
I believe it is better to use the Icon whenever we have an Icon at other siblings components because we gain with all the other features that are already implemented at the Icon component like `size`, `fill`, `stroke`, `spacing` ...

Reply:
In this context, IconStyled is a Button that contains a "dummy" SVG icon, but if we use the Icon we could define the size and margins at itself, together with all the other design rules that could come over at the future

Reply:
@marcosricardo and @ggdaltoso, do you think that the gain is good enough to explain the effort that will take to changing the approach?

Reply:
> but if we use the Icon we could define the size and margins at itself, together with all the other design rules that could come over at the future

I don't see this as a gain at all. I believe that one shouldn't be able to customize the Icon inside, since one of the premises of the design system is to be somehow opinionated on how one can use its elements. Letting the internal Icon customizable could lead other teams to make UI that shouldn't be possible since they don't follow Yoga principles.

Unlike Icon within Button, the container itself (our Button), allows system usage as the default button. An Icon container in a button shouldn't (IMO) be as "customizable" as an individual Icon component

Does this makes sense for you folks?

Reply:
I don't know if I write it wrong but what I meant was that the yoga developers could customize the Icon itself using the Icon props instead of using the svg inside the StyledIcon, that way we have more control and the gain to have all the Icon features to be used at the ButtonIcon.

Reply:
Sorry, @ericcleao I misunderstood your suggestion. I definitely agree with this suggestion, this refactor will make the code more accurate with how we expected someone to use Yoga with system.